### PR TITLE
chore(kit): lint files

### PR DESCRIPTION
**Pull request by DamianGlowala**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR lints files within `package/kit/` with an internal set of ESLint configs and (temporarily) excludes this directory from being linted using the current settings to avoid CI errors. It is the first part of a transition to a currently developed library based on flat config.

Linting the project resulted in more consistent and richer JSDoc comments with links to relevant resources in the docs, more readable code, fixing minor bugs and was also a great opportunity to remove some DRY code (see e.g. added `toArray` util) or do little refactors of the code.

The matter of which rules should be enabled/disabled is certainly subject to change (including their configurations). I am currently trying to find this 'sweet spot' for each of the configs used to lint the Nuxt repo.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/context.ts [hunk 1]
```diff
@@ -1,40 +1,42 @@
 import { getContext } from 'unctx'
-import type { Nuxt } from '@nuxt/schema'
+import { type Nuxt } from '@nuxt/schema'
 
 /** Direct access to the Nuxt context - see https://github.com/unjs/unctx. */
-export const nuxtCtx = getContext<Nuxt>('nuxt')
-
-// TODO: Use use/tryUse from unctx. https://github.com/unjs/unctx/issues/6
+export const nuxtContext = getContext<Nuxt>('nuxt')
```
Comment (danielroe) [inline packages/kit/src/context.ts:5]: We likely need to keep this export the same or it is a breaking change.

File: packages/kit/src/internal/cjs.ts [hunk 2]
```diff
@@ -15,106 +16,133 @@ export interface ResolveModuleOptions {
...
   clearCache?: boolean
 
   /** Automatically de-default the result of requiring the module. */
   interopDefault?: boolean
 }
 
+// eslint-disable-next-line jsdoc/require-param, jsdoc/require-returns
 /** @deprecated Do not use CJS utils */
-function isNodeModules (id: string) {
+function isNodeModules(id: string) {
```
Comment (danielroe) [inline packages/kit/src/internal/cjs.ts:31]: can we keep the space before brackets? 🙏

File: packages/kit/src/context.ts (hunk 1 above)
Comment (DamianGlowala) [reply packages/kit/src/context.ts]: Added `nuxtCtx` to the allowlist of `unicorn/prevent-abbreviations`.

File: packages/kit/src/internal/cjs.ts (hunk 2 above)
Comment (DamianGlowala) [reply packages/kit/src/internal/cjs.ts]: Done :)

File: packages/kit/src/build.ts [hunk 3]
```diff
@@ -1,135 +1,168 @@
...
   const nuxt = useNuxt()
 
-  if (options.dev === false && nuxt.options.dev) {
-    return
-  }
-  if (options.build === false && nuxt.options.build) {
+  if (
+    (options.dev === false && nuxt.options.dev)
+    || (options.build === false && !nuxt.options.dev)
+  ) {
```
Comment (danielroe) [inline packages/kit/src/build.ts:56]: I think this is clearer:


```suggestion
  if (options.dev === false && nuxt.options.dev) {
    return
  }
  if (options.build === false && nuxt.options.build) {
```

File: packages/kit/src/build.ts [hunk 4]
```diff
@@ -1,135 +1,168 @@
...
+
       if (config) {
-        fn(config)
+        callbackFunction(config)
       }
-    }
-    if (options.client !== false) {
-      const config = configs.find(i => i.name === 'client')
+    } else if (options.client !== false) {
+      const config = configs.find((index) => index.name === 'client')
```
Comment (danielroe) [inline packages/kit/src/build.ts:68]:
```suggestion
      const config = configs.find((config) => config.name === 'client')
```

File: packages/kit/src/build.ts [hunk 5]
```diff
@@ -1,135 +1,168 @@
...
+    (options.dev === false && nuxt.options.dev)
+    || (options.build === false && !nuxt.options.dev)
+  ) {
     return
   }
 
   nuxt.hook('webpack:config', (configs: WebpackConfig[]) => {
     if (options.server !== false) {
-      const config = configs.find(i => i.name === 'server')
+      const config = configs.find((index) => index.name === 'server')
```
Comment (danielroe) [inline packages/kit/src/build.ts:62]:
```suggestion
      const config = configs.find((config) => config.name === 'server')
```

File: packages/kit/src/build.ts [hunk 6]
```diff
@@ -1,135 +1,168 @@
...
   const nuxt = useNuxt()
 
-  if (options.dev === false && nuxt.options.dev) {
-    return
-  }
-  if (options.build === false && nuxt.options.build) {
+  if (
+    (options.dev === false && nuxt.options.dev)
+    || (options.build === false && !nuxt.options.dev)
+  ) {
```
Comment (danielroe) [inline packages/kit/src/build.ts:92]:
```suggestion
  if (options.dev === false && nuxt.options.dev) {
    return
  }
  if (options.build === false && nuxt.options.build) {
```

File: packages/kit/src/build.ts [hunk 7]
```diff
@@ -1,135 +1,168 @@
...
     return
   }
 
   if (options.server !== false && options.client !== false) {
-    // Call fn() only once
-    return nuxt.hook('vite:extend', ({ config }) => fn(config))
+    // Call callbackFunction() only once
+    nuxt.hook('vite:extend', ({ config }) => {
+      callbackFunction(config)
+    })
```
Comment (danielroe) [inline packages/kit/src/build.ts:100]: It's important the promise is awaited.

```suggestion
    nuxt.hook('vite:extend', ({ config }) => callbackFunction(config))
```

File: packages/kit/src/build.ts [hunk 8]
```diff
@@ -1,135 +1,168 @@
...
-    if (options.server !== false && isServer) {
-      return fn(config)
-    }
-    if (options.client !== false && isClient) {
-      return fn(config)
+    if (
+      (options.server !== false && isServer)
+      || (options.client !== false && isClient)
+    ) {
+      callbackFunction(config)
```
Comment (danielroe) [inline packages/kit/src/build.ts:110]:
```suggestion
    if (options.server !== false && isServer) {
      return callbackFunction(config)
    }
    if (options.client !== false && isClient) {
      return callbackFunction(config)
```

File: packages/kit/src/resolve.ts [hunk 9]
```diff
@@ -35,9 +42,9 @@ export async function resolvePath (path: string, opts: ResolvePathOptions = {}):
 
   // Use current nuxt options
   const nuxt = tryUseNuxt()
-  const cwd = opts.cwd || (nuxt ? nuxt.options.rootDir : process.cwd())
-  const extensions = opts.extensions || (nuxt ? nuxt.options.extensions : ['.ts', '.mjs', '.cjs', '.json'])
-  const modulesDir = nuxt ? nuxt.options.modulesDir : []
+  const cwd = options.cwd || (nuxt ? nuxt.options.rootDir : process.cwd())
+  const extensions = options.extensions || (nuxt ? nuxt.options.extensions : ['.ts', '.mjs', '.cjs', '.json'])
+  const modulesDirectory = nuxt ? nuxt.options.modulesDir : []
```
Comment (danielroe) [inline packages/kit/src/resolve.ts:47]: Maybe better: `nodeModulesDirectories`?

File: .eslintrc [hunk 10]
```diff
@@ -4,7 +4,8 @@
     "dist",
     "public",
     "node_modules",
-    "packages/schema/schema"
+    "packages/schema/schema",
+    "packages/kit"
```
Comment (danielroe) [inline .eslintrc:8]: Can we revert the linting for now and just include the JSdoc + readability improvements? 🙏

File: packages/kit/src/build.ts [hunk 11]
```diff
@@ -55,25 +59,29 @@ export function extendWebpackConfig (
...
+      const config = configs.find((config) => config.name === 'server')
+
       if (config) {
-        fn(config)
+        callbackFunction(config)
       }
-    }
-    if (options.client !== false) {
-      const config = configs.find(i => i.name === 'client')
+    } else if (options.client !== false) {
```
Comment (danielroe) [inline packages/kit/src/build.ts:67]: this is a behaviour change here

File: packages/kit/src/build.ts [hunk 12]
```diff
@@ -86,27 +94,42 @@ export function extendViteConfig (
   }
 
   if (options.server !== false && options.client !== false) {
-    // Call fn() only once
-    return nuxt.hook('vite:extend', ({ config }) => fn(config))
+    // Call callbackFunction() only once
+    nuxt.hook('vite:extend', ({ config }) => callbackFunction(config))
```
Comment (danielroe) [inline packages/kit/src/build.ts:98]: if we don't return the promise we need to await it

File: packages/kit/src/resolve.ts [hunk 13]
```diff
@@ -150,17 +194,35 @@ export async function resolveNuxtModule (base: string, paths: string[]) {
...
 }
 
-export async function resolveFiles (path: string, pattern: string | string[], opts: { followSymbolicLinks?: boolean } = {}) {
-  const files = await globby(pattern, { cwd: path, followSymbolicLinks: opts.followSymbolicLinks ?? true })
-  return files.map(p => resolve(path, p)).filter(p => !isIgnored(p)).sort()
+export async function resolveFiles (
+  path: string,
+  pattern: string | string[],
+  options: { followSymbolicLinks?: boolean } = {}
+) {
```
Comment (danielroe) [inline packages/kit/src/resolve.ts:217]: can we revert the lint change that splits function parameters into separate lines? 🙏

File: packages/kit/src/resolve.ts [hunk 14]
```diff
@@ -82,65 +96,95 @@ export async function resolvePath (path: string, opts: ResolvePathOptions = {}):
...
   }
+
   return _resolveAlias(path, alias)
 }
 
 export interface Resolver {
-  resolve (...path: string[]): string
-  resolvePath (path: string, opts?: ResolvePathOptions): Promise<string>
+  resolve(...path: string[]): string
+  resolvePath(path: string, options?: ResolvePathOptions): Promise<string>
```
Comment (danielroe) [inline packages/kit/src/resolve.ts:146]:
```suggestion
  resolve (...path: string[]): string
  resolvePath (path: string, options?: ResolvePathOptions): Promise<string>
```

**Comment by danielroe** [comment]
I think we can now close this as we've migrated to the new nuxt/eslint-config, and also it's badly merge conflicted.

Feel free to open a new PR with updates/improvements that may have been missed 🙏

**Comment by DamianGlowala** [comment]
I might find some time to port the JSDoc comments which would definitely bring value :)

**Comment by danielroe** [comment]
That would be amazing 🙏

**Pull request closed without merging**